tests/unit/src/test/scala/tests/DebugDiscoverySuite.scala

```diff
@@ -0,0 +1,253 @@
+package tests
+
+import java.util.concurrent.TimeUnit
+
+import scala.meta.internal.metals.DebugDiscoveryParams
+import scala.meta.internal.metals.JsonParser._
+import scala.meta.internal.metals.debug.BuildTargetContainsNoMainException
+import scala.meta.internal.metals.debug.NoTestsFoundException
+import scala.meta.internal.metals.debug.WorkspaceErrorsException
+
+// note(@tgodzik) all test have `System.exit(0)` added to avoid occasional issue due to:
+// https://stackoverflow.com/questions/2225737/error-jdwp-unable-to-get-jni-1-2-environment
+class DebugDiscoverySuite extends BaseDapSuite("debug-discovery") {
+  private val mainPath = "a/src/main/scala/a/Main.scala"
+  private val fooPath = "a/src/main/scala/a/Foo.scala"
+  private val barPath = "a/src/main/scala/a/Bar.scala"
+
+  test("run") {
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{
+           |  "a": {}
+           |}
+           |/${mainPath}
+           |package a
+           |object Main {
+           |  def main(args: Array[String]) = {
+           |    print("oranges are nice")
+           |    System.exit(0)
+           |  }
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen(mainPath)
+      _ <- server.waitFor(TimeUnit.SECONDS.toMillis(10))
+      debugger <- server.startDebuggingUnresolved(
+        new DebugDiscoveryParams(
+          s"file:///${server.toPath(mainPath)}",
```

Comment:
Why not do `server.toPath(mainPath).toURI.toString` ?

Reply:
ugh... out of all the things I tried... how did I not try this. This worked. Thank you 🙏🏼